### PR TITLE
ui: Add proxy cache config UI to org settings (PROJQUAY-7697)

### PR DESCRIPTION
**Pull request by harishsurf**
Adds proxy cache config UI to organization settings tab

proxy-cache-config

Follow up:
- show proxy cache configuration during org creation (separate PR)

### CODE REVIEW

**Review by bcaton85 — CHANGES_REQUESTED** [review]
Overall looks good, functionally correct. There are a couple of bugs and additional tests are required but nothing major.

File: web/src/resources/ProxyCacheResource.ts [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+  const proxyResponse = await axios.get(proxyCacheConfigUrl, {signal});
+  assertHttpCode(proxyResponse.status, 200);
+  return proxyResponse.data;
+}
+
+export async function validateProxyCacheConfig(
+  proxyCacheConfig: IProxyCacheConfig,
+) {
+  const proxyCacheConfigUrl = `/api/v1/organization/${proxyCacheConfig.org_name}/validateproxycache`;
+  const proxyResponse = await axios.post(proxyCacheConfigUrl, proxyCacheConfig);
```
[the remark this answers is not in the document]
Comment (harishsurf) [reply web/src/resources/ProxyCacheResource.ts:17]: Hmm, for me it works fine when in incognito mode. In normal browser mode, these fields get autofilled and clearing them sets these fields to an empty string. I am not seeing an option to disable autofill on either `TextInput` or `Form` PF component. Hence, I have added the check for empty string in the above code right before `axios.post`

File: web/src/routes/OrganizationsList/Organization/Tabs/Settings/Settings.tsx [hunk 2]
```diff
@@ -48,6 +49,17 @@ export default function Settings(props: SettingsProps) {
...
+    {
+      name: 'Proxy-Cache config',
+      id: 'proxycacheconfig',
+      content: (
+        <ProxyCacheConfig
+          organizationName={props.organizationName}
+          isUser={props.isUserOrganization}
+        />
+      ),
+      visible: quayConfig?.features?.PROXY_CACHE,
```
[the remark this answers is not in the document]
Comment (harishsurf) [reply web/src/routes/OrganizationsList/Organization/Tabs/Settings/Settings.tsx:61]: good catch! added a test

File: web/src/resources/ProxyCacheResource.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (bcaton85) [reply web/src/resources/ProxyCacheResource.ts]: okay makes sense. As long as there's a condition that if the fields are empty string they should be set to null when being sent in the request that should be fine

**Review by bcaton85 — APPROVED** [review]
LGTM - this will need to be cherrypicked into redhat-3.13

**Pull request merged**